cmd/api/src/api/v2/auth/oidc.go

```diff
@@ -42,17 +42,15 @@ func (s ManagementResource) CreateOIDCProvider(response http.ResponseWriter, req
 		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, err.Error(), request), response)
 	} else if validated := validation.Validate(createRequest); validated != nil {
 		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, validated.Error(), request), response)
-	} else if strings.Contains(createRequest.Name, " ") {
-		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, "invalid name formatting, ensure there are no spaces in the provided name", request), response)
 	} else {
 		var (
-			formattedName = strings.ToLower(createRequest.Name)
+			formattedName = strings.ToLower(strings.ReplaceAll(createRequest.Name, " ", "-"))
```

Comment:
What if we moved the slug generation into the SSO create db method instead? That way it's isolated to just what needs it and other callers don't need to know to implement it in a specific way.